src/node_main_instance.cc

```diff
@@ -0,0 +1,160 @@
+#include "node_main_instance.h"
+#include "node_internals.h"
+#include "node_options-inl.h"
+#include "node_v8_platform-inl.h"
+
+namespace node {
+
+using v8::Context;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::Locker;
+using v8::SealHandleScope;
+
+NodeMainInstance::NodeMainInstance(uv_loop_t* event_loop,
+                                   const std::vector<std::string>& args,
+                                   const std::vector<std::string>& exec_args)
+    : args_(args),
+      exec_args_(exec_args),
+      array_buffer_allocator_(CreateArrayBufferAllocator()),
+      isolate_(nullptr),
+      isolate_data_(nullptr) {
+  // TODO(joyeecheung): when we implement snapshot integration this needs to
+  // set params.external_references.
+  Isolate::CreateParams params;
+  params.array_buffer_allocator = array_buffer_allocator_.get();
+  isolate_ =
+      NewIsolate(&params, event_loop, per_process::v8_platform.Platform());
+  CHECK_NOT_NULL(isolate_);
+  isolate_data_.reset(CreateIsolateData(isolate_,
```

Comment:
When implementing snapshot this needs to be changed to something like `DeserializeIsolateData()` which uses `GetDataFromSnapshotOnce()` (instead of doing e.g. `String::NewFromOneByte`).